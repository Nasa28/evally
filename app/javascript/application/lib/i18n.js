const messages = {
  en: {
    buttons: {
      archive: 'Archive',
      cancel: 'Cancel',
      complete: 'Complete',
      copy: 'Copy',
      create: 'Create',
      delete: 'Delete',
      edit: 'Edit',
      hide: 'Hide',
      show: 'Show'
    },
    destroy_modals: {
      default: {
        title: 'Delete',
        confirm: 'Are you sure you want to delete it?'
      },
      evaluation: {
        title: 'Delete evaluation',
        confirm: 'Are you sure you want to delete this evaluation?'
      },
      employee: {
        title: 'Delete employee',
        confirm: 'Are you sure you want to delete this employee?'
      },
      template: {
        title: 'Delete template',
        confirm: 'Are you sure you want to delete this template?'
      }
    },
    employees: {
      title: 'Employees',
      buttons: {
        add_new: 'New employee',
        permalink: 'Permalink',
        edit: 'Edit employee',
        archive: 'Archive employee',
        delete: 'Delete employee'
      },
      flashes: {
        archive: {
          success: 'Employee has been succefully archived.',
          error: 'Employee cannot be archived due to some error: {reason}'
        },
        create: {
          success: 'Employee has been succefully created.',
          error: 'Employee cannot be created due to some error: {reason}'
        },
        edit: {
          success: 'Employee has been succefully updated.',
          error: 'Employee cannot be updated due to some error: {reason}'
        },
        fetch: {
          error: 'Employees cannot be loaded due to some error: {reason}'
        },
        delete: {
          success: 'Evaluation has been succefully deleted.',
          error: 'Evaluation cannot be destroyed due to some error: {reason}'
        }
      },
      forms: {
        archive: {
          title: 'Archive employee',
          warning: 'This action cannot be undo. All drafted evaluations of the employee will be destroyed!',
          text: 'Please, provide the reason to archive the employee: ',
          too_experienced: 'Too experienced',
          released: 'Released',
          released_at: 'Released at'
        },
        permalink: {
          title: 'Share permalink',
          text: 'Copy the permalink below to share the latest evaluation:'
        },
        create_title: 'Create employee',
        edit_title: 'Edit employee',
        first_name: 'First name',
        last_name: 'Last name',
        position: 'Position',
        hired_at: 'On board since',
        next_review: 'Next review'
      },
      messages: {
        select: 'Select employee to see his latest evaluation.',
        not_evaluated: '{name} has not been evaluated yet.'
      },
      sidebar: {
        search: 'Search ...',
        subheader: 'List of employees',
        no_items: 'There are no employees'
      }
    },
    evaluations: {
      title: 'Evaluations',
      buttons: {
        add_new: 'New evaluation',
        complete: 'Complete evaluation',
        save_draft: 'Save draft',
        reset: 'Undo changes',
        delete: 'Delete evaluation'
      },
      flashes: {
        create: {
          success: 'Evaluation has been succefully created.',
          error: 'Evaluation cannot be created due to some error: {reason}'
        },
        fetch: {
          error: 'Evaluation cannot be loaded due to some error: {reason}'
        },
        save: {
          success: 'Evaluation has been succefully updated.',
          error: 'Evaluation cannot be updated due to some error: {reason}'
        },
        complete: {
          success: 'Evaluation has been succefully completed.',
          error: 'Evaluation cannot be completed due to some error: {reason}'
        },
        delete: {
          success: 'Evaluation has been succefully deleted.',
          error: 'Evaluation cannot be destroyed due to some error: {reason}'
        }
      },
      forms: {
        complete: {
          title: 'Complete evaluation',
          question: 'Are you sure you want to complete this evaluation?',
          warning: 'This action cannot be undo.',
          next_label: 'Next evaluation at'
        },
        create_title: 'Create draft',
        select_employee: 'Select employee',
        select_template: 'Select template',
        employee_label: 'Employee',
        template_label: 'Template',
        bool_no: 'No',
        bool_yes: 'Yes',
        hired_at: 'hired at',
        next_review: 'next review',
        completed_at: 'completed at'
      },
      message: 'Select draft evaluation or create a new one.',
      sidebar: {
        search: 'Search ...',
        subheader: 'Evaluations drafts',
        subtitle: 'based on \'{name}\' template',
        no_items: 'There are no evaluations drafts'
      }
    },
    dashboard: {
      title: 'Hello {name}!',
      common: {
        as: ' as ',
        first_time: 'First time'
      },
      drafts: {
        title: 'Drafts',
        edit_tooltip: 'Edit',
        updated: 'Updated {date}'
      },
      upcoming: {
        title: 'Upcoming',
        new_tooltip: 'Create'
      },
      activities: {
        title: 'Activities',
        no_activities: 'There are no activities today',
        see_more: 'See more'
      }
    },
    navbar: {
      dashboard: 'Dashboard',
      evaluations: 'Evaluations',
      employees: 'Employees',
      templates: 'Templates',
      archive: 'Archive',
      settings: 'Settings',
      logout: 'Logout'
    }
  },

  pl: {
    buttons: {
      archive: 'Archiwizuj',
      cancel: 'Anuluj',
      complete: 'Zakończ',
      copy: 'Skopiuj',
      create: 'Utwórz',
      delete: 'Usuń',
      edit: 'Edytuj',
      hide: 'Ukryj',
      show: 'Pokaż'
    },
    destroy_modals: {
      default: {
        title: 'Usuń',
        confirm: 'Czy na pewno chcesz to usunąć?'
      },
      evaluation: {
        title: 'Usuń ocenę',
        confirm: 'Czy na pewno chcesz usunąć ocenę tego pracownika?'
      },
      employee: {
        title: 'Usuń pracownika',
        confirm: 'Czy na pewno chcesz usunąć tego pracownika?'
      },
      template: {
        title: 'Usuń szablon',
        confirm: 'Czy na pewno chcesz usunąć ten szablon?'
      }
    },
    employees: {
      title: 'Pracownicy',
      buttons: {
        add_new: 'Nowy pracownik',
        permalink: 'Permalink',
        edit: 'Edytuj pracownika',
        archive: 'Przenieś do archiwum',
        delete: 'Usuń pracownika'
      },
      forms: {
        archive: {
          title: 'Przenieś do archiwum',
          warning: 'Nie ma możliwości cofnięcia tej akcji. Wszystkie szkice ocen tego pracownika zostaną usunięte!',
          text: 'Proszę podać powód przeniesienia pracownika do archiwum: ',
          too_experienced: 'Bogate doświadczenie',
          released: 'Koniec współpracy',
          released_at: 'Współpraca zakończona'
        },
        permalink: {
          title: 'Udostępnij link',
          text: 'Skopiuj link z dostępem do aktualnej oceny tego pracownika:'
        },
        create_title: 'Utwórz pracownika',
        edit_title: 'Edytuj pracownika',
        first_name: 'Imię',
        last_name: 'Nazwisko',
        position: 'Stanowisko',
        hired_at: 'Zatrudniony od',
        next_review: 'Następne review'
      },
      flashes: {
        archive: {
          success: 'Pracownik został pomyślnie zarchiwizowany.',
          error: 'Wystąpił błąd podczas archiwizowania pracownika: {reason}'
        },
        create: {
          success: 'Pracownik został pomyślnie dodany.',
          error: 'Wystąpił błąd podczas dodawania pracownika: {reason}'
        },
        edit: {
          success: 'Pracownik został pomyślnie zapisany.',
          error: 'Wystąpił błąd podczas zapisywania pracownika: {reason}'
        },
        fetch: {
          error: 'Wystąpił błąd podczas wczytywania pracowników: {reason}'
        },
        delete: {
          success: 'Pracownik został pomyślnie usunięty.',
          error: 'Wystąpił błąd podczas usuwania pracownika: {reason}'
        }
      },
      messages: {
        select: 'Wybierz pracownika z listy, aby zobaczyć jego aktualną ocenę.',
        not_evaluated: '{name} nie został jeszcze oceniony.'
      },
      sidebar: {
        search: 'Szukaj ...',
        subheader: 'Lista pracowników',
        no_items: 'Brak pracowników'
      }
    },
    evaluations: {
      title: 'Oceny',
      buttons: {
        add_new: 'Nowa ocena',
        complete: 'Zakończ ocenę',
        save_draft: 'Zapisz szkic',
        reset: 'Cofnij zmiany',
        delete: 'Usuń ocenę'
      },
      flashes: {
        create: {
          success: 'Ocena została pomyślnie utworzona.',
          error: 'Wystąpił błąd podczas tworzenia szkicu: {reason}'
        },
        fetch: {
          error: 'Wystąpił błąd podczas wczytywania ocen: {reason}'
        },
        save: {
          success: 'Ocena została pomyślnie zapisana.',
          error: 'Wystąpił błąd podczas zapisu: {reason}'
        },
        complete: {
          success: 'Ocena została pomyślnie zakończona.',
          error: 'Wystąpił błąd podczas zapisu: {reason}'
        },
        delete: {
          success: 'Ocena została pomyślnie usunięta.',
          error: 'Wystąpił błąd podczas usuwania: {reason}'
        }
      },
      forms: {
        complete: {
          title: 'Zakończ ocenę',
          question: 'Czy na pewno chcesz zakończyć ocenę pracownika?',
          warning: 'Ta akcja jest ostateczna i nie można jej cofnąć.',
          next_label: 'Następne review'
        },
        create_title: 'Utwórz szkic',
        select_employee: 'Wybierz pracownika',
        select_template: 'Wybierz szablon',
        employee_label: 'Pracownik',
        template_label: 'Szablon',
        bool_no: 'Nie',
        bool_yes: 'Tak',
        hired_at: 'zatrudniono',
        next_review: 'następne review',
        completed_at: 'zakończono'
      },
      message: 'Wybierz szkic oceny z listy lub stwórz nowy.',
      sidebar: {
        search: 'Szukaj ...',
        subheader: 'Szkice ocen',
        subtitle: 'według szablonu \'{name}\'',
        no_items: 'Brak szkiców ocen'
      }
    },
    dashboard: {
      title: 'Cześć {name}!',
      common: {
        as: ' jako ',
        first_time: 'Pierwszy raz'
      },
      drafts: {
        title: 'Szkice',
        edit_tooltip: 'Edytuj',
        updated: 'Edytowano {date}'
      },
      upcoming: {
        title: 'Nadchodzące',
        new_tooltip: 'Utwórz'
      },
      activities: {
        title: 'Aktywności',
        no_activities: 'Brak aktywności',
        see_more: 'Więcej'
      }
    },
    navbar: {
      dashboard: 'Kokpit',
      evaluations: 'Oceny',
      employees: 'Pracownicy',
      templates: 'Szablony',
      archive: 'Archiwum',
      settings: 'Ustawienia',
      logout: 'Wyloguj'
    }
  }
}

export default messages