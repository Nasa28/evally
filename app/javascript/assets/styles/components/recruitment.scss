.recruitment {
  &-row {
    &__header {
      display: flex;
      justify-content: space-between;
      align-items: flex-start;
    }

    &__title {
      font-size: rem2dec(2.4rem);
      font-weight: normal;
      line-height: 120%;
      margin-bottom: rem2dec(2rem);
    }

    &__date {
      font-size: rem2dec(1.8rem);
      color: $font-gray-light;
    }

    &__description {
      font-size: rem2dec(1.4rem);
      color: $font-gray-dark;
    }

    &__participants {
      & > * {
        margin-right: rem2dec(.5rem)
      }
    }
  }

  &-participant {
    font-size: rem2dec(1rem);
    font-weight: bold;
    color: $white;
  }

  &-stages {
    width: 100%;
    padding: rem2dec(1.6rem 0 1rem);
    overflow-x: auto;
    display: flex;
    justify-content: flex-start;
    align-items: stretch;
    flex: 1 0 auto;

    @include scrollbar-styling();
  }

  &-stage {
    min-width: 100%;
    min-height: rem2dec(12rem);
    border: 1px solid $grey;
    border-radius: rem2dec(.5rem);

    @include respond(md) {
      min-width: 50%;
    };

    @include respond(lg) {
      min-width: 33%;
    };

    @include respond(xl) {
      min-width: 25%;
    };

    &:not(:last-of-type) {
      margin-right: rem2dec(1rem);
    }

    &:only-child {
      width: 100%;

      @include respond(md) {
        width: 50%;
      };

      @include respond(lg) {
        width: 33%;
      };

      @include respond(xl) {
        width: 25%;
      };

    }

    &__title {
      font-size: rem2dec(1.8rem);
      font-weight: normal;
      line-height: 120%;
      padding: rem2dec(1.76rem 2rem);
      text-align: center;
      border-bottom: 1px solid $grey;
      position: relative;
      display: flex;
      justify-content: center;
      align-items: baseline;
    }

    .sortable-ghost {
      border: 1px dashed $grey;
    }

    &__content {
      height: calc(100% - rem2dec(5.7rem));
      max-height: calc(100vh - rem2dec(15rem));
      overflow-y: auto;

      @include scrollbar-styling();
    }

    &__actions {
      display: flex;
      margin-left: rem2dec(1rem);
    }
  }

  &-candidate {
    margin: rem2dec(1rem);
    border: 1px solid $grey;
    border-radius: rem2dec(.5rem);
    display: flex;
    flex-direction: column;

    &__body {
      padding: rem2dec(1rem);
      display: flex;
      justify-content: space-between;
      align-items: flex-start;
    }

    &__footer {
      padding: rem2dec(.5rem 1rem);
      border-top: 1px solid $grey;
      display: flex;
      justify-content: space-around;
      align-items: center;
    }

    &__info {
      flex: 1 1 auto;
    }

    &__name {
      font-size: rem2dec(1.4rem);
      display: flex;
      align-items: center;

      & > a {
        text-decoration: none;
      }
    }

    &__details {
      font-size: rem2dec(1.2rem);
      color: $font-gray-dark;
    }
  }
}
