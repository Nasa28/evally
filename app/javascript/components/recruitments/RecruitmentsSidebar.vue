<template>
  <div class="sidebar sidebar--boxed">
    <div class="vcard">
      <h3 class="vcard__fullname">{{ recruitDocument.fullname }}</h3>

      <div class="vcard__content">
        <div class="vcard__info">
          <div class="vcard__label">{{ $t('components.recruitments.sidebar.gender') }}</div>
          <div class="vcard__value">{{ recruitDocument.gender }}</div>
        </div>
        <div class="vcard__info">
          <div class="vcard__label">{{ $t('components.recruitments.sidebar.email') }}</div>
          <div class="vcard__value">{{ recruitDocument.email }}</div>
        </div>
        <div class="vcard__info">
          <div class="vcard__label">{{ $t('components.recruitments.sidebar.phoneNumber') }}</div>
          <div class="vcard__value">{{ recruitDocument.phone }}</div>
        </div>
        <div class="vcard__info">
          <div class="vcard__label">{{ $t('components.recruitments.sidebar.group') }}</div>
          <div class="vcard__value">{{ recruitDocument.group }}</div>
        </div>
        <div class="vcard__info">
          <div class="vcard__label">{{ $t('components.recruitments.sidebar.position') }}</div>
          <div class="vcard__value">{{ recruitDocument.position }}</div>
        </div>
        <div class="vcard__info">
          <div class="vcard__label">{{ $t('components.recruitments.sidebar.received_at') }}</div>
          <div class="vcard__value">{{ recruitDocument.receivedAt }}</div>
        </div>
      </div>
    </div>

  </div>
</template>

<script>
import { RecruitDocument } from '@models/recruit_document'

export default {
  name: 'RecruitmentsSidebar',
  props: {
    recruitDocument: {
      type: RecruitDocument,
      required: true,
      default: () => new RecruitDocument()
    }
  }
}
</script>
