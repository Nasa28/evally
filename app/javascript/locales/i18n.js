import Vue from 'vue'
import VueI18n from 'vue-i18n'

Vue.use(VueI18n)

import en from '@locales/en'
import pl from '@locales/pl'

export default new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages: {
    en,
    pl
  }
})
