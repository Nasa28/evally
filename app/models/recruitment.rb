# frozen_string_literal: true

class Recruitment < ApplicationRecord
  belongs_to :candidate

  # # Validation
  #
  validates :group, :position, :received_at, :status, presence: true

  # # Enums
  #
  enum status: {
    imported: 'imported',
    fresh: 'fresh',
    ongoing: 'ongoing',
    awaiting_response: 'awaiting_response',
    suspended: 'suspended',
    accepted: 'accepted',
    rejected: 'rejected'
  }
end
