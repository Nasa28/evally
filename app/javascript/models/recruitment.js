import { Model, List } from './base'
import i18n from '@locales/i18n'

class Recruitment extends Model {
  get defaults() {
    return {
      id: null,
      candidate_id: null,
      gender: '',
      first_name: '',
      last_name: '',
      email: '',
      status: '',
      group: '',
      phone: '',
      position: '',
      received_at: '',
      social_links: '',
      source: ''
    };
  }

  get fullname() {
    return [this.first_name, this.last_name].join(' ')
  }

  static get routes() {
    return {
      recruitmentsPath: '/v2/recruitments',
      recruitmentPath: id => `/v2/recruitments/${id}`
    }
  }
}

class RecruitmentsList extends List {
  get model() {
    return Recruitment;
  }
}

export { Recruitment, RecruitmentsList };
