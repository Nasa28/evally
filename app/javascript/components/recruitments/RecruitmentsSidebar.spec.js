import { shallowMount } from '@vue/test-utils'
import i18n from '@locales/i18n'

import RecruitmentsSidebar from './RecruitmentsSidebar'

import { Recruitment } from '@models/recruitment'

describe('RecruitmentsSidebar', () => {
  const props = {
    recruitment: new Recruitment({email: 'e@domain.org'}),
  }

  const wrapper = shallowMount(
    RecruitmentsSidebar,
    {
      propsData: props,
      stubs: ['v-layout', 'v-flex', 'v-text-field'],
      mocks: {
        $t: key => i18n.t(key)
      }
    })

  it('has a name', () => {
    expect(RecruitmentsSidebar.name).toEqual('RecruitmentsSidebar')
  })

  it('props', () => {
     expect(wrapper.props().recruitment.email).toEqual('e@domain.org')
   })

  it('matches do snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
