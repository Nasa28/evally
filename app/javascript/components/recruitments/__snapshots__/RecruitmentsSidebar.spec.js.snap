// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecruitmentsSidebar matches do snapshot 1`] = `
<div class="sidebar sidebar--boxed">
  <div class="vcard">
    <h3 class="vcard__fullname"> </h3>
    <div class="vcard__content">
      <div class="vcard__info">
        <div class="vcard__label">Gender</div>
        <div class="vcard__value"></div>
      </div>
      <div class="vcard__info">
        <div class="vcard__label">Email</div>
        <div class="vcard__value">e@domain.org</div>
      </div>
      <div class="vcard__info">
        <div class="vcard__label">Phone number</div>
        <div class="vcard__value"></div>
      </div>
      <div class="vcard__info">
        <div class="vcard__label">Group</div>
        <div class="vcard__value"></div>
      </div>
      <div class="vcard__info">
        <div class="vcard__label">Position</div>
        <div class="vcard__value"></div>
      </div>
      <div class="vcard__info">
        <div class="vcard__label">Received at</div>
        <div class="vcard__value">Feb 10, 2020 00:00</div>
      </div>
    </div>
  </div>
</div>
`;
